luigi/contrib/s3.py

```diff
@@ -604,27 +527,20 @@ def __init__(self, path, s3_client, **kwargs):
         self.s3_options = kwargs
 
     def move_to_final_destination(self):
-        self.s3_client.put_multipart(self.tmp_path, self.path, **self.s3_options)
+        self.s3_client.put_multipart(
+            self.tmp_path, self.path, **self.s3_options)
 
 
 class ReadableS3File(object):
     def __init__(self, s3_key):
-        self.s3_key = s3_key
+        # This is a botocore StreamingBody object
+        self.s3_key = s3_key.get()['Body']
```

Comment:
@dlstadther 
http://botocore.readthedocs.io/en/latest/reference/response.html#botocore.response.StreamingBody
This is not iterable :(